docs/core-concepts/app-deployment.md

```diff
@@ -1,31 +1,283 @@
+        {
+            public static void Main()
+            {
+                 Console.WriteLine("Enter any text, followed by <Enter>:\n");
+                 String s = Console.ReadLine();
+                 ShowWords(s);
+                 Console.Write("\nPress any key to continue... ");
+                 Console.ReadKey();
+          }
+
+          private static void ShowWords(String s)
+          {
+              String pattern = @"\w+";
+              var matches = Regex.Matches(s, pattern);
+              if (matches.Count == 0)
+                  Console.WriteLine("\nNo words were identified in your input.");
+              else
+              {
+                  Console.WriteLine("\nThere are {0} words in your string:", matches.Count);
+                  for (int ctr = 0; ctr < matches.Count; ctr++)
+                      Console.WriteLine("   #{0,2}: '{1}' at position {2}", ctr,
+                                        matches[ctr].Value, matches[ctr].Index);
+              }
+          }
+      }
+    }
+    ```
+
+3. Run the `dotnet restore` [dotnet restore](https://docs.microsoft.com/en-us/dotnet/core/core-concepts/core-sdk/cli/dotnet-restore "dotnet restore") command to restore the dependencies specified in your project.
+
+4. Create a debug build of your app by using the [dotnet build](https://docs.microsoft.com/en-us/dotnet/core/core-concepts/core-sdk/cli/dotnet-build "dotnet build") command.
+
+5. After you've debugged and tested the program, you can create the files to be deployed with your app by using the [dotnet publish -f netcoreapp1.0 -c release](https://docs.microsoft.com/en-us/dotnet/core/core-concepts/core-sdk/cli/dotnet-build "dotnet publish -f netcoreapp1.0 -c release") command. This creates a release (rather than a debug) version of your app.
+
+   The resulting files are placed in a directory named `publish` that is in a subdirectory of your project's
+`.\bin\release\netcoreapp1.0` subdirectory.
+
+6. Along with your application's files, the publishing process emits a program database (.pdb) file that contains debugging information about your app. The file is useful primarily for debugging exceptions; you can choose not to package it with your application's files.
+
+The complete set of application  files can be deployed in any way you'd like. For example, you can package them in a zip file, use a simple `copy` command, or deploy them with any installation package of your choice.
+
+Before deploying your app, you can also user `crossgen` to convert it to native code. However, its performance impact is smaller than for self-contained apps. For more information, see the [Native Image Generation](#crossgen) section.
+
+In addition to the application binaries the installer should also either bundle the shared framework installer or check for it as a pre-requisite as part of the installation of the application.  Installation of the shared framework requires Administrator/root access since it is machine wide. 
```

Comment:
nits: 
- missing comma after application binaries
- pre-requisite -> prerequisite
- as part of the installation of the application -> as part of the application installation